File: backend/src/main/java/com/romans/backend/model/Order.java
package com.romans.backend.model;

import jakarta.persistence.*;
import java.time.LocalDateTime;
import java.util.*;

@Entity
@Table(name = "orders")
public class Order {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private LocalDateTime createdAt = LocalDateTime.now();
  private LocalDateTime deliveredAt = LocalDateTime.now();

  @ManyToOne
  @JoinColumn(name = "user_id")
  private User client;

  @OneToOne
  @JoinColumn(name = "user_id")
  private User courier;

  @Enumerated(EnumType.STRING)
  private OrderStatus status = OrderStatus.CONFIRMED;

  @OneToMany(mappedBy = "order", cascade = CascadeType.ALL)
  private List<OrderItem> items = new ArrayList<>();

  @Enumerated(EnumType.STRING)
  private DeliveryType deliveryType = DeliveryType.DELIVERY;

  private String deliveryAddress;
}

File: frontend/src/types/order.ts
import { DeliveryType, OrderStatus } from "./enums";
import type { Dish } from "./dish";
import type { User } from "./user";
import type { Address } from "./address";

export interface OrderItem {
  id: number;
  priceAtMoment: number;
  quantity: number;
  dish: Dish;
}

export interface Order {
  id: number;
  createdAt: string;
  deliveredAt?: string;
  deliveryAddress: Address;
  deliveryType: DeliveryType;
  totalPrice: number;
  status: OrderStatus;
  client: Pick<User, "id" | "name">;
  courier?: Pick<User, "id" | "name">;
  items: OrderItem[];
}
